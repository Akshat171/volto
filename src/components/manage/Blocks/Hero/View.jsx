/**
 * View hero block.
 * @module components/manage/Blocks/Hero/View
 */

import React from 'react';
import { withBlockExtensions } from '@plone/volto/helpers';
import { HeroBody } from '@plone/volto/components';
import cx from 'classnames';

export const View = (props) => {
  const { data } = props;
  return (
    <div
      className={cx(
        'block hero align',
        {
          // we keep the default alignment of the original hero block
          left: !Boolean(data.align),
        },
        data.align,
      )}
    >
      <HeroBody {...props}></HeroBody>
    </div>
  );
};

export default withBlockExtensions(View);
