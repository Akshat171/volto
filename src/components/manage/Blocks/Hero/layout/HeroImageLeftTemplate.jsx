import React from 'react';
import {
  HeroImage,
  HeroText,
} from '@plone/volto/components/manage/Blocks/Hero/components';

const HeroImageLeftTemplate = (props) => (
  <div className="block-inner-wrapper">
    <HeroImage {...props} />
    <HeroText {...props} />
  </div>
);

export default HeroImageLeftTemplate;
