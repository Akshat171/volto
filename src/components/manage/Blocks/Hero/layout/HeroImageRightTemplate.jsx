import React from 'react';
import {
  HeroImage,
  HeroText,
} from '@plone/volto/components/manage/Blocks/Hero/components';

const HeroImageRightTemplate = (props) => (
  <div className="block-inner-wrapper">
    <HeroText {...props} />
    <HeroImage {...props} />
  </div>
);

export default HeroImageRightTemplate;
