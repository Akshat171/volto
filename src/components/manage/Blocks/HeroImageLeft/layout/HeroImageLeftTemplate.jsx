import React from 'react';
import {
  HeroImage,
  HeroText,
  HeroToolbar,
} from '@plone/volto/components/manage/Blocks/HeroImageLeft/components';

const HeroImageLeftTemplate = (props) => (
  <>
    <HeroToolbar {...props} className="toolbar-left" />
    <HeroImage {...props} />
    <HeroText {...props} />
  </>
);

export default HeroImageLeftTemplate;
