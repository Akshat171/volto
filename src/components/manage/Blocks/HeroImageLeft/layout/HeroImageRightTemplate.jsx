import React from 'react';
import {
  HeroImage,
  HeroText,
  HeroToolbar,
} from '@plone/volto/components/manage/Blocks/HeroImageLeft/components';

const HeroImageRightTemplate = (props) => (
  <>
    <HeroToolbar {...props} className="toolbar-right" />
    <HeroText {...props} />
    <HeroImage {...props} />
  </>
);

export default HeroImageRightTemplate;
