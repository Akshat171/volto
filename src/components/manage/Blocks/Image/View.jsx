/**
 * View image block.
 * @module components/manage/Blocks/Image/View
 */

import React from 'react';
import PropTypes from 'prop-types';
import { UniversalLink } from '@plone/volto/components';
import cx from 'classnames';
import Image from '@plone/volto/components/theme/Image/Image';

/**
 * View image block class.
 * @class View
 * @extends Component
 */
const View = ({ data, detached }) => (
  <p
    className={cx(
      'block image align',
      {
        center: !Boolean(data.align),
        detached,
      },
      data.align,
    )}
  >
    {data.url && (
      <>
        {(() => {
          const image = (
            <Image
              image={data.url}
              alt={data.alt}
              className={cx({
                'full-width': data.align === 'full',
                large: data.size === 'l',
                medium: data.size === 'm',
                small: data.size === 's',
              })}
            />
          );
          if (data.href) {
            return (
              <UniversalLink
                href={data.href}
                openLinkInNewTab={data.openLinkInNewTab}
              >
                {image}
              </UniversalLink>
            );
          } else {
            return image;
          }
        })()}
      </>
    )}
  </p>
);

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
View.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default View;
