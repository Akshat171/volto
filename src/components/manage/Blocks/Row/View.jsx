import React from 'react';
import PropTypes from 'prop-types';
import { Grid } from 'semantic-ui-react';
import cx from 'classnames';
import { RenderBlocks } from '@plone/volto/components';
import { withBlockExtensions } from '@plone/volto/helpers';

const RowBlockView = (props) => {
  const { data } = props;
  const metadata = props.metadata || props.properties;
  const columns = data.data.blocks_layout.items;

  return (
    <div
      className={cx('block __grid', {
        [data['@type']]: data['@type'] !== '__grid',
        centered: data.align === 'center' || data.align === undefined,
        'space-between': data.align === 'space-between',
        'centered-text': data.centeredText,
        one: columns?.length === 1,
        two: columns?.length === 2,
        three: columns?.length === 3,
        four: columns?.length === 4,
      })}
    >
      {data.headline && <h2 className="headline">{data.headline}</h2>}

      <Grid stackable stretched columns={columns.length}>
        <RenderBlocks
          {...props}
          blockWrapperTag={Grid.Column}
          metadata={metadata}
          content={data.data}
        />
      </Grid>
    </div>
  );
};

/**
 * Property types.
 * @property {Object} propTypes Property types.
 * @static
 */
RowBlockView.propTypes = {
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default withBlockExtensions(RowBlockView);
