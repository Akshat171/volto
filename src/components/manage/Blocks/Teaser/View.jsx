import React from 'react';
import TeaserBody from './Body';
import { withBlockExtensions } from '@plone/volto/helpers';

const TeaserBlockView = (props) => {
  return <TeaserBody {...props} />;
};

export default withBlockExtensions(TeaserBlockView);
