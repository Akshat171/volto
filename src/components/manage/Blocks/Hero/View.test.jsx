import React from 'react';
import renderer from 'react-test-renderer';
import { View } from './View';
import { Provider } from 'react-intl-redux';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import config from '@plone/volto/registry';

import HeroImageLeftTemplate from '@plone/volto/components/manage/Blocks/Hero/layout/HeroImageLeftTemplate';
import HeroImageRightTemplate from '@plone/volto/components/manage/Blocks/Hero/layout/HeroImageRightTemplate';

const mockStore = configureStore();

const store = mockStore({
  intl: {
    locale: 'en',
    messages: {},
  },
});

const blockId = '1234';

config.blocks.blocksConfig = {
  hero: {
    id: 'hero',
    title: 'Hero',
    group: 'media',
    extensions: {},
    variations: [
      {
        id: 'heroImageLeft',
        title: 'Image on left side',
        view: HeroImageLeftTemplate,
        isDefault: true,
      },
      {
        id: 'hereImageRight',
        title: 'Image on right side',
        view: HeroImageRightTemplate,
        isDefault: false,
      },
    ],
    restricted: false,
    mostUsed: true,
    sidebarTab: 1,
    security: {
      addPermission: [],
      view: [],
    },
  },
};

test('renders a view hero component', () => {
  const component = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <View
          data={{ url: 'heroimage.jpg', '@type': 'hero' }}
          block={blockId}
          variation={{}}
        />
      </MemoryRouter>
    </Provider>,
  );

  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
