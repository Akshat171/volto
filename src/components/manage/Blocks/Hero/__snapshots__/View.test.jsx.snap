// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a view hero component 1`] = `
<div
  className="block hero"
>
  <div
    className="block-inner-wrapper"
  >
    <div
      className="image-upload-widget-image"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="toolbar"
    >
      <div
        className="image-upload-widget-toolbar"
      >
        <div
          className="ui buttons"
        >
          <button
            className="ui basic icon button"
            onClick={[Function]}
          >
            <svg
              className="icon circled"
              dangerouslySetInnerHTML={
                Object {
                  "__html": undefined,
                }
              }
              onClick={null}
              style={
                Object {
                  "fill": "#e40166",
                  "height": "24px",
                  "width": "auto",
                }
              }
              viewBox=""
              xmlns=""
            />
          </button>
        </div>
      </div>
      <img
        alt=""
        className="hero-image"
        src="heroimage.jpg/@@images/image/teaser"
      />
    </div>
    <div
      className="hero-body"
    >
      <div
        className="hero-text"
      />
    </div>
  </div>
</div>
`;
